components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/SessionMgtDBQueries.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.authentication.framework.util;
+
+/**
+ * This class contains default SQL queries.
+ */
+public class SessionMgtDBQueries {
+
+    // LOAD Queries
+    private static final String LOAD_SESSION_STORE_DATA = "SELECT SESSION_ID, MAX(TIME_CREATED) AS TIME_CREATED " +
+            "FROM IDN_AUTH_SESSION_STORE WHERE SESSION_TYPE = ''AppAuthFrameworkSessionContextCache'' " +
+            "AND EXPIRY_TIME > ? AND (TENANT_ID = ? OR TENANT_ID = -1){0} GROUP BY SESSION_ID";
+            // TODO: remove TENANT_ID=-1 from clause when SessionDataStore.removeSessionData() had been updated
+            //       to store session's actual tenant id. (https://github.com/wso2/product-is/issues/12210)
```

Comment:
Is this `todo` intentional?

Reply:
Yes. As explained in Git issue, tenant id is not persisted when "deleting" a session and as a workaround we had to put an OR clause to search by the default value (TENANT_ID = -1). Once the Git issue had been addressed this OR clause should be deleted and the TODO could be removed.